zerver/lib/markdown/__init__.py

```diff
@@ -1876,10 +1876,12 @@ def handleMatch(  # type: ignore[override] # supertype incompatible with supersu
                     self.md.zulip_rendering_result.mentions_wildcard = True
                 user_id = "*"
             elif user:
+                assert type(user) == FullNameInfo
```

Comment:
This should be an `isinstance` check.

Reply:
I don't think it really matters since we're never gonna subclass `FullNameInfo`.

Reply:
The right idiom when it doesn’t matter is `isinstance`, according to PEP 8.

Reply:
One curiosity here is that I need to add the type check here in order for mypy to complain that I had had `user["full_name"]` instead of `user.full_name`.  If you convert the `if user` check to `if user is not None` then mypy still can't apparently infer that `user` is a type of `FullNameInfo`.  Fortunately, the test suite quickly alerted me to the code that I needed to change.

Anyway, I am converting the assertion to use `isinstance`.  (Either flavor of assertion helps mypy produce an error along the lines of "Value of type "FullNameInfo" is not indexable" if you fail to cut over to attribute access.)

Reply:
(I also changed the check from `elif user` to `ellif user is not None`.)